packages/dds/map/src/test/mocha/map.spec.ts

```diff
@@ -773,6 +777,133 @@ describe("Map", () => {
+			let map2: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				containerRuntimeFactory = new MockContainerRuntimeFactory();
+				// Create the first map
+				map1 = createConnectedMap("map1", containerRuntimeFactory, options);
+				// Create and connect a second map
+				map2 = createConnectedMap("map2", containerRuntimeFactory, options);
+			});
+
+			it("Can generate proper attribution in connected state", () => {
+				map1.set("first", "value1");
+				map1.set("second", "value2");
+				map2.set("first", "value3");
+
+				containerRuntimeFactory.processSomeMessages(1);
+
+				assert.notEqual(
+					map1.getAttribution("first"),
+					undefined,
+					"map1 did not generate correct attribution",
+				);
+
+				containerRuntimeFactory.processAllMessages();
+
+				assert.deepEqual(
+					map1.getAllAttribution(),
+					map2.getAllAttribution(),
+					"map1 and map2 do not have consistent attribution",
+				);
+
+				map1.clear();
+				containerRuntimeFactory.processAllMessages();
+				assert.equal(
+					map1.getAllAttribution()?.size,
+					0,
+					"the attribution of map1 was not cleared",
+				);
+				assert.equal(
+					map2.getAllAttribution()?.size,
+					0,
+					"the attribution of map2 was not cleared",
+				);
+			});
+		});
+
+		describe("Summarization", () => {
+			let containerRuntimeFactory: MockContainerRuntimeFactory;
+			let map1: SharedMap;
+			let map2: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				containerRuntimeFactory = new MockContainerRuntimeFactory();
+				// Create the first map
+				map1 = createConnectedMap("map1", containerRuntimeFactory, options);
+				// Create and connect a second map
+				map2 = createConnectedMap("map2", containerRuntimeFactory, options);
+			});
+
+			it("Can summarize/load the map correctly with attribution", async () => {
+				map1.set("first", "value1");
+				map1.set("second", "value2");
+
+				containerRuntimeFactory.processAllMessages();
+
+				const summaryContent = (map1.getAttachSummary().summary.tree.header as ISummaryBlob)
+					.content;
+
+				assert.equal(
+					map1.getAllAttribution()?.size,
+					2,
+					"do not have correct number of attribution",
+				);
+				assert.equal(
+					summaryContent,
+					`{"blobs":[],"content":{"first":{"type":"Plain","value":"value1","attribution":{"type":"op","seq":1}},"second":{"type":"Plain","value":"value2","attribution":{"type":"op","seq":2}}}}`,
+					"the serialized data is not consistent",
+				);
+
+				const services = new MockSharedObjectServices({ header: summaryContent });
+				await map2.load(services);
+
+				assert.equal(map1.get("first"), map2.get("first"));
+				assert.equal(map1.get("second"), map2.get("second"));
+				assert.deepEqual(
+					map1.getAllAttribution(),
+					map2.getAllAttribution(),
+					"map1 and map2 do not have consistent attribution",
+				);
+			});
```

Comment:
since there are some nontrivial changes to the blobbing logic of map, I think it's worth adding test coverage for larger summaries to be confident everything is working correctly.

Reply:
Resolved by 812089c8300df2692a0b26c0b0f82c7e19bc3c5f